modules/sharedaddy/sharing.js

```diff
@@ -1,364 +1,474 @@
+		}.bind( this );
 
-			$more_sharing_pane
-				.css( {
-					left: $more_sharing_button.position().left + 'px',
-					top: $more_sharing_button.position().top + $more_sharing_button.height() + 3 + 'px',
-				} )
-				.slideToggle( 200 );
-		} );
+		this.paneLeave = function () {
+			if ( this.openedBy === 'hover' ) {
+				this.resetCloseTimer();
+			}
+		}.bind( this );
+
+		this.documentClick = function () {
+			this.close();
+		}.bind( this );
+
+		this.button.addEventListener( 'click', this.buttonClick );
+		document.addEventListener( 'click', this.documentClick );
 
 		if ( document.ontouchstart === undefined ) {
 			// Non-touchscreen device: use hover/mouseout with delay
-			$more_sharing_buttons.hover(
-				function () {
-					var $more_sharing_button = $( this ),
-						$more_sharing_pane = $more_sharing_button.parents( 'div:first' ).find( '.inner' ),
-						timer;
-
-					if ( ! $more_sharing_pane.is( ':animated' ) ) {
-						// Create a timer to make the area appear if the mouse hovers for a period
-						timer = setTimeout( function () {
-							var handler_item_leave,
-								handler_item_enter,
-								handler_original_leave,
-								handler_original_enter,
-								close_it;
-
-							$sharing_email.slideUp( 200 );
-
-							$more_sharing_pane.data( 'justSlid', true );
-							$more_sharing_pane
-								.css( {
-									left: $more_sharing_button.position().left + 'px',
-									top:
-										$more_sharing_button.position().top + $more_sharing_button.height() + 3 + 'px',
-								} )
-								.slideDown( 200, function () {
-									// Mark the item as have being appeared by the hover
-									$more_sharing_button.data( 'hasoriginal', true ).data( 'hasitem', false );
-
-									setTimeout( function () {
-										$more_sharing_pane.data( 'justSlid', false );
-									}, 300 );
-
-									$more_sharing_pane
-										.mouseleave( handler_item_leave )
-										.mouseenter( handler_item_enter );
-									$more_sharing_button
-										.mouseleave( handler_original_leave )
-										.mouseenter( handler_original_enter );
-								} );
+			this.button.addEventListener( 'mouseenter', this.buttonEnter );
+			this.button.addEventListener( 'mouseleave', this.buttonLeave );
+			this.pane.addEventListener( 'mouseenter', this.paneEnter );
+			this.pane.addEventListener( 'mouseleave', this.paneLeave );
+		}
+	};
 
-							// The following handlers take care of the mouseenter/mouseleave for the share button and the share area - if both are left then we close the share area
-							handler_item_leave = function () {
-								$more_sharing_button.data( 'hasitem', false );
+	// ---------------------------- SHARE COUNTS ---------------------------- //
 
-								if ( $more_sharing_button.data( 'hasoriginal' ) === false ) {
-									var timer = setTimeout( close_it, 800 );
-									$more_sharing_button.data( 'timer2', timer );
-								}
-							};
+	if ( window.sharing_js_options && window.sharing_js_options.counts ) {
```

Comment:
Presumably `sharing_js_options` is set by `wp_localize_script`. If `sharing_js_options` is empty, then I _imagine_ we shouldn't load any of this JS at all. Let's move that condition into PHP so we don't load any JS that we aren't going to run.

Reply:
The way the script was written, it makes me think that `sharing_js_options` is only needed for the sharing counts, and not for the rest of the sharing button functionality that the script handles (more button, print button, email button, etc). I don't really know if that's the case, though, or if it's just a coding mistake in the first place.